api/src/convert/input_ring.rs

```diff
@@ -0,0 +1,101 @@
+use std::convert::{TryFrom, TryInto};
+
```

Comment:
`std::convert::{TryFrom, TryInto}` are part of the 2021 edition prelude, so they can be omitted 
```suggestion
```